08.ls_object/long_format.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class LongFormat
+  def initialize(items)
+    @items = items
+  end
+
+  def create_text
+    total = "total #{sum_blocks(@items)}"
+    table = @items.empty? ? [] : create_table(@items)
```

Comment:
```suggestion
    table = create_table(@items)
```

のままにして、`create_table`が`[]`を返すようにした方がスッキリするな〜と思いました。

Reply:
> `create_table`が`[]`を返す

早速ありがとうございます🙏 修正してみました！ 3621587
（意味を取り違えてたらすみません）

`create_table`が`[]`を返す方がスッキリする理由について、よく分かっていないので教えていただけますでしょうか🙇‍♂️
アイテム数がいくつかによって処理を分けるのはあくまで内部的な都合なので、パブリックメソッド側で気にするよりもプライベートメソッドで面倒を見た方が良いから、という考え方で合っていますか？

Reply:
> という考え方で合っていますか？

うーん、僕の考えとは微妙に違うかも？
僕の考えはこんな感じでした。

- `create_table`に「テーブルを作ってね！」とお願いしたら、こっちは何も考えないで戻り値を受け取るようにしたい
- `@items.empty? ? [] :`は`@items`があれば条件分岐できるので、`create_table`側に持っていって判断するのも簡単そう
- 「`create_table`の引数には空の配列を渡してはならない」という制約があるより、「俺が何でも受け止めるからどーんと来い！」っていうメソッドの方が気軽に呼びやすい（呼び出し側が気を遣う、というのは避けたい）

[この記事](https://qiita.com/jnchito/items/49fe82364bca2f4f37f5)にも書いたのですが、「使う側に気を遣わせる」設計ではなく、 「作る側が気を遣う」設計の方が基本的に良いと思います。

Reply:
詳しい説明をありがとうございます🙏✨
> 「使う側に気を遣わせる」設計ではなく、 「作る側が気を遣う」設計

という考え方がまだ徹底できていないなぁと分かりました😅
これからもコードを書くときに気にしてみたいと思います

また、ご提案いただいた早期リターンの書き方に修正しました 1067027